README.md

```diff
@@ -33,26 +33,81 @@ Please open an issue if something is out of date.
 
 ## Usage
 
-This is what I use in my configuration. It should work regardless of if you've
-cloned this overlay locally.
+There are two ways to use this overlay on NixOS, depending on which
+nixpkgs channel you follow.
+
+### Usage (nixos-unstable)
+
+This usage just utilizes [`overlay` functionality from `nixpkgs`]().
+
+Note that when using the overlay, the module will automatically reference the correct
+`sway-beta` package, just as the environment packages come from `pkg`.
+
+```nix
+{ config, lib, pkgs, ... }:
+let
+  nos = "https://github.com/colemickens/nixpkgs-wayland/archive/master.tar.gz";
+  swayOverlay = (import (builtins.fetchTarball nos));
+in
+  {
+    nixpkgs.overlays = [ swayOverlay ];
+    programs.sway.enable = true;
+    environment.systemPackages = with pkgs; [
+      # DO NOT list `sway-beta` here! It will override the module!
+      # sway-beta
+
+      # essentials for any wlroots compositor
+      grim slurp mako wlstream redshift-wayland
+
+      # optional alternative bar
+      waybar
+    ];
+  }
+```
+
+### Usage (nixos-18.09)
+
+Rather than maintaining an unknown number of backports from nixos-unstable to here
+for supporting `nixos-18.09` users, these instructions will pull in this overlay's
+packages without using an actual proper overlay.
+
+Note that when *not* using the overlay, the module must be told which `sway-beta` to use
+just as the environment packages must come from `swaypkgs`.
 
 ```nix
-{ ... }:
+{ config, lib, pkgs, ... }:
 let
   nos = "https://github.com/colemickens/nixpkgs-wayland/archive/master.tar.gz";
-  swayOverlay =
-    if builtins.pathExists /etc/nixpkgs-wayland
-    then (import /etc/nixpkgs-wayland)
-    else (import (builtins.fetchTarball nos));
+  swayOverlay = (import (builtins.fetchTarball nos));
+  nixpkgs = import (builtins.fetchTarball {
+    url = "https://github.com/NixOS/nixpkgs-channels/archive/nixos-unstable.tar.gz";
+  });
+  swaypkgs = (import nixpkgs {
+    config = config.nixpkgs.config;
+    overlays = [ swayOverlay ];
+  }).pkgs;
```

Comment:
A simpler version, this has the advantage that swaypkgs controls the nixpkgs version and so should be more compatible with the binary-cache:
```suggestion
  swaypkgs = import "${builtins.fetchTarball nos}/build.nix";
```